datafusion/src/logical_plan/expr.rs

```diff
@@ -2189,6 +2189,20 @@ pub fn exprlist_to_fields<'a>(
     expr.into_iter().map(|e| e.to_field(input_schema)).collect()
 }
 
+/// Calls a named built in function
+/// ```
+/// use datafusion::logical_plan::*;
+///
+/// // create the expression sin(x) < 0.2
+/// let expr = call_builtin_scalar_fn("sin", vec![col("x")]).unwrap().lt(lit(0.2));
+/// ```
+pub fn call_builtin_scalar_fn(name: impl AsRef<str>, args: Vec<Expr>) -> Result<Expr> {
+    match name.as_ref().parse::<functions::BuiltinScalarFunction>() {
+        Ok(fun) => Ok(Expr::ScalarFunction { fun, args }),
+        Err(e) => Err(e),
+    }
```

Comment:
i wonder if it makes sense to make it a macro rather than a function call so that nonexisteng built-in functions will be caught during compile time not runtime

Reply:
I agree, `let expr = call_builtin_scalar_fn!(ToTimestamp, vec![lit("2020-09-08T12:00:00+00:00")])` is as readable.

Reply:
Agree with you @Jimexist ! 

And alamb 's opinion of the trade-off between macro and function call is here: 
https://github.com/apache/arrow-datafusion/issues/1718#issuecomment-1026239024

Reply:
Is it needed to implement both?

Reply:
No I don't think so. If we want a macro we can always add that as a follow on PR. 

Thanks @HaoYang670  !